clang/test/Driver/clang-offload-wrapper-multi-batch.c

```diff
@@ -0,0 +1,214 @@
+
+// Generate sym files
+// RUN: echo '_entry_targ_1_AA'                                              >  %t.1-A.sym
+// RUN: echo '_entry_targ_1_BB'                                              >  %t.1-B.sym
+// RUN: echo '_entry_targ_1_CC'                                              >  %t.1-C.sym
+
+// Generate table file
+// RUN: echo '[Code|Properties|Symbols]'                                     >  %t.1.table
+// RUN: echo '%t.1A.bin|%t.1-A.prop|%t.1-A.sym'                              >> %t.1.table
+// RUN: echo '%t.1B.bin|%t.1-B.prop|%t.1-B.sym'                              >> %t.1.table
+// RUN: echo '%t.1C.bin|%t.1-C.prop|%t.1-C.sym'                              >> %t.1.table
+
+///////////////////////////////////////////////////////////////////////////////////////////
+// Generate input and table for target 2
+///////////////////////////////////////////////////////////////////////////////////////////
+
+// Generate image files
+// RUN: echo 't2-A' > %t.2A.bin
+// RUN: echo 't2-B' > %t.2B.bin
+// RUN: echo 't2-C' > %t.2C.bin
+
+// Generate property files
+// RUN: echo '[testProp-GroupG]'                                             >  %t.2-A.prop
+// RUN: echo 'testProp-name17=1|0'                                           >> %t.2-A.prop
+
+// RUN: echo '[testProp-GroupH]'                                             >  %t.2-B.prop
+// RUN: echo 'testProp-name19=1|0'                                           >> %t.2-B.prop
+
+// RUN: echo '[testProp-GroupI]'                                             >  %t.2-C.prop
+// RUN: echo 'testProp-name23=1|0'                                           >> %t.2-C.prop
+
+// Generate sym files
+// RUN: echo '_entry_targ_2_AAA'                                             >  %t.2-A.sym
+// RUN: echo '_entry_targ_2_BBB'                                             >  %t.2-B.sym
+// RUN: echo '_entry_targ_2_CCC'                                             >  %t.2-C.sym
+
+// Generate table file
+// RUN: echo '[Code|Properties|Symbols]'                                     >  %t.2.table
+// RUN: echo '%t.2A.bin|%t.2-A.prop|%t.2-A.sym'                              >> %t.2.table
+// RUN: echo '%t.2B.bin|%t.2-B.prop|%t.2-B.sym'                              >> %t.2.table
+// RUN: echo '%t.2C.bin|%t.2-C.prop|%t.2-C.sym'                              >> %t.2.table
+
+///////////////////////////////////////////////////////////////////////////////////////////
+// Generate wrapped BC file with multiple targets using multiple batch (i.e. table) files
+// and generate executable
+///////////////////////////////////////////////////////////////////////////////////////////
+// RUN: clang-offload-wrapper "-o=%t.wrapped.bc"           \
+// RUN:   -kind=sycl -target=target-zero -batch %t.0.table \
+// RUN:   -kind=sycl -target=target-one  -batch %t.1.table \
+// RUN:   -kind=sycl -target=target-two  -batch %t.2.table
+// RUN: %clang -fdeclspec %s %t.wrapped.bc -o %t.fat.bin
```

Comment:
what is a purpose of this line?

Reply:
The images are later extracted from %t.fat.bin, and the extracted images are checked to make sure they are the same as the images from the table inputs.

Reply:
I am interested in `-fdeclspec` command line option.
According to the [doc](https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-fdeclspec)
```
-fdeclspec, -fno-declspec
Allow __declspec as a keyword
```

Why do you need `__declspec` keyword?

Reply:
I based this test on clang-offload-extract.c, which uses -fdeclspec, perhaps unnecessarily.  I'll remove it from my test.